src/pages/rewards/MigratePanel.tsx

```diff
@@ -0,0 +1,220 @@
+import styled, { AnyStyledComponent } from 'styled-components';
+import { useSelector } from 'react-redux';
+
+import { ENVIRONMENT_CONFIG_MAP } from '@/constants/networks';
+import { STRING_KEYS } from '@/constants/localization';
+import { ButtonAction, ButtonSize, ButtonType } from '@/constants/buttons';
+
+import { useAccountBalance, useBreakpoints, useStringGetter } from '@/hooks';
+
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { Details } from '@/components/Details';
+import { Panel } from '@/components/Panel';
+import { Button } from '@/components/Button';
+import { Link } from '@/components/Link';
+import { IconName, Icon } from '@/components/Icon';
+import { Output, OutputType } from '@/components/Output';
+import { VerticalSeparator } from '@/components/Separator';
+import { Tag } from '@/components/Tag';
+import { WithReceipt } from '@/components/WithReceipt';
+
+import { getSelectedNetwork } from '@/state/appSelectors';
+
+const TOKEN_MIGRATION_LEARN_MORE_LINK =
+  'https://www.dydx.foundation/blog/update-on-exploring-the-future-of-dydx';
+
+export const MigratePanel = () => {
+  const { isNotTablet } = useBreakpoints();
+  const stringGetter = useStringGetter();
+
+  const selectedNetwork = useSelector(getSelectedNetwork);
+
+  const chainId = Number(ENVIRONMENT_CONFIG_MAP[selectedNetwork].ethereumChainId);
+
+  // v3 token is only on mainnet
+  const { balance: tokenBalance } = useAccountBalance({
+    addressOrDenom: chainId === 1 ? import.meta.env.VITE_V3_TOKEN_ADDRESS : undefined,
+    chainId: 1,
+    isCosmosChain: false,
+  });
+
+  return isNotTablet ? (
+    <Styled.MigratePanel
+      slotHeader={<Styled.Title>{stringGetter({ key: STRING_KEYS.MIGRATE })}</Styled.Title>}
+      slotRight={
+        <Styled.MigrateAction>
+          <div>
+            <div>{stringGetter({ key: STRING_KEYS.AVAILABLE_TO_MIGRATE })}</div>
+            <Styled.Token type={OutputType.Asset} value={tokenBalance} />
+          </div>
+          {import.meta.env.VITE_TOKEN_MIGRATION_URI && (
+            <Button
+              action={tokenBalance ? ButtonAction.Primary : ButtonAction.Base}
```

Comment:
I noticed `tokenBalance` is a string, and comes back as `'0'` when balance is 0, which is truthy, and does not disabled the button.

Do we also need to set if tokenBalance is 0/undefined `isDisabled`?

Reply:
will convert to number!
re `isDisabled`, no we are still going to let user click migrate now to visit the migration UI regardless of balance